modules/cdp-engine/src/lib.rs

```diff
@@ -506,6 +500,41 @@ pub mod module {
 }
 
 impl<T: Config> Pallet<T> {
+	fn accumulate_interest(now: Moment, last_accumulate_timestamp: Moment) -> u32 {
+		let mut count: u32 = 0;
+
+		if !T::EmergencyShutdown::is_shutdown() && !now.is_zero() {
+			let interval = now.saturating_sub(last_accumulate_timestamp);
+			let multiplier = Rate::checked_from_rational(interval, ONE_YEAR).unwrap_or_default();
+
+			for currency_id in T::CollateralCurrencyIds::get() {
+				let annual_interest_rate = Self::get_annual_interest_rate(currency_id);
```

Comment:
Should have some tests to verify that running this for a year will actually result something similar to target rate to ensure we don't have too much precision lose due to arithmetic